### ISSUE TITLE
Atlassian has decided not to implement syntax highlighting in new PR UI

### ISSUE THREAD

**Issue by RobertChrist** [refined-bitbucket/refined-bitbucket#376]
tl;dr: Atlassian dropped syntax highlighting from the new PR UI roadmap, would you reconsider the feature or accept a PR on this?

-------

Hi @reyronald, in Issue #252 , you stated that syntax highlighting (and I assume other related features, like double-clicking a word to highlight all occurrences of the word) will not be rolled out to the new PR UI, because native support was on Atlassian's roadmap.

Well, its been a year, and in that year, Atlassian has decided to drop this item from its roadmap.  https://jira.atlassian.com/browse/BCLOUD-8673 moved from 'In Progress' back to 'Under Consideration' in June, with no progress since.

Would you be open to implementing this feature, or be willing to accept a pull request from myself that does so?

**Comment by reyronald** [comment]
Hey @RobertChrist 

Yes, given Atlassian's difficulties with delivering this, I'll give it a shot myself.

Someone in that issue you referenced provided a temporary work-around in case you're interested and it works for you:  https://gist.github.com/lephuongbg/bf26e0aa24b3c9b0071b7597522ded15

**Comment by RobertChrist** [comment]
Yup, I'm already running that greasemonkey script.  

Okay, that's great to hear!  I very much look forward to this patch.  Please let me know if there's any way I can be helpful

**Comment by pingcheng** [comment]
Does anyone know why Atlassian is not doing the colour highlighting?
Looks like this is a feature that can purely do on client's side.

**Comment by RobertChrist** [comment]
@pingcheng No idea, but from nearly a decade of working with Atlassian products, this seems like normality for them.

I just opened a PR to add syntax highlight, and syntax theme support to this plugin, so it may be available soon, or [you can try it out in my repo](https://github.com/RobertChrist/refined-bitbucket/tree/feature/RC/AddSyntaxHighlightInNewUI_AddSyntaxHighlightThemeOptions) in the meanwhile.  HTH

**Comment by RobertChrist** [comment]
@reyronald Hi Rey!  My team officially swapped over to bitbucket.org two days ago, so I went ahead and implemented this feature.

I also decided to add syntax highlighting theme support, and made it backwards compatible for the old diff/PR UIs.

PR available here: https://github.com/refined-bitbucket/refined-bitbucket/pull/386

Let me know if you need anything, or any changes you would like me to make?  Or any other ways I can help?  Tis a fun extension to play with.

**Comment by littleGnAl** [comment]
Any update on this?

**Comment by RobertChrist** [comment]
@littleGnAl still sitting in PR waiting for @reyronald to review and merge

**Comment by chrisaligent** [comment]
The new PR experience will be made mandatory from Feb 15th onwards:
![Selection_1806](https://user-images.githubusercontent.com/45612883/103495627-58131b80-4e8b-11eb-9a47-3845f836d53e.png)

I can't perform code reviews without syntax highlighting, so if there's anything I can do to help get this reviewed/merged please let me know. I don't want to pressure anyone but the impact of losing this feature greatly concerns me, and I'm sure if it isn't merged by Feb 15th we'll probably see a spike in GitHub issues asking why syntax highlighting is "broken".

I've asked Atlassian to hold off forcing us onto the new PR experience until there's either first-party or third-party support available, but I'm not expecting a response (and it seems very unlikely that they'll suddenly implement this by that date, or even publicly acknowledge that this amazing browser extension exists).

**Comment by RobertChrist** [comment]
Hey @chrisaligent , I agree.  I'm not sure what the hold up is, I'm still waiting on a PR review / merge from @reyronald or @andremw

**Comment by triforcely** [comment]
Hey @RobertChrist, I tested your feature based on extension built off _feature/RC/AddSyntaxHighlightInNewUI_AddSyntaxHighlightThemeOptions_ branch and I've found some issues with it:

1. When navigating to comment from the "Activity" tab it causes a full page refresh after windows scrolls to the desired area.
![image](https://user-images.githubusercontent.com/12202969/103545283-a16e6580-4ea1-11eb-9608-a15d152ec74d.png)
2. "View entire file" modal window doesn't have syntax highlighting at all.

Looks like a good starting point for the feature though :)

**Comment by RobertChrist** [comment]
@triforcely Hi Triforcely,
    I haven't received any update from the repository owner, @reyronald, since opening this PR last November.  As such, I'm not really sure what to do with this PR.    He did review another one of my PR's available here: https://github.com/refined-bitbucket/refined-bitbucket/pull/387, but by the time the PR was reviewed, I was in the process of accepting a new position in my career,, and as such, haven't had time to follow through with the update yet.
    
So... I'm not sure what to do here.  I would like to spend the effort to address the issue you have found, but am not keen to do so if the PR is simply going to languish here.  Additionally, Atlassian is dropping support for the old UI entirely in mid February according to @chrisaligent above.  That means about half of the code in this repository will need to be deleted.
  
  So to summarize, my life has become unexpectedly busy since November, I haven't had any feedback on this PR by the repository maintainer, and this repository is going to need a major overhaul in order to stay relevant in about ~2 weeks time.  This all leaves me very hesitant to make updates at the moment, and unsure what the correct course of action is.

**Comment by andremw** [comment]
Hey guys, I've been talking to @reyronald about this for some time since it's getting harder and harder for him to maintain the extension, and I think that the time has come to address this matter.

@reyronald, what would we need to do in order to bring more contributors? How can I help?

Would someone here be interested in becoming a maintainer with Ronald?

We would need to move the deployment to Github Actions (or any other CI/CD tool) so that we can set the secret keys there and enable any maintainer to deploy a new version by pushing to a specific branch. I can help with this if needed.

**Comment by reyronald** [comment]
Everyone can relax, this is still on my radar.

At the time this pull request was opened there was a backlog of other existing PRs that needed review first that I've been slowly clearing since. This one is the most recent to have been created and therefore took the last spot of my todo list.

It's also been low priority since Atlassian contacted me directly to let me know they were working on this internally, and it wasn't until recently that it was announced that the old UI was shutting down. I've had calls with the Bitbucket team and I am sure that not having this implemented natively by this point was not the original plan at all, not sure why it has proven so intensily difficult for them to deliver. 

Because of that I was hesitant to have a community-implemented alternative that could potentially cause an internal de-prioritization of the native implementation, which is highly likely given their awareness of this project.

As for now, I am aware of the new February deadline and also that it is the most useful feature of the extension. I will take care of getting a version of the feature working in the new UI by then, so not need to worry.

@RobertChrist , don't feel that you need to address the issue commented above. I can take care of getting this to completion.

**Comment by andremw** [comment]
We hope so!

Let me know if you need any help.

**Comment by andremw** [comment]
We hope so.

Let me know if you need any help.

**Comment by andremw** [comment]
@reyronald, we hope so.

Let me know if you need any help.

**Comment by RobertChrist** [comment]
@reyronald This is great to hear!  Thank you for the update, I was very confused.  That makes a lot of sense!  Thank you very much!

**Comment by RobertChrist** [comment]
@triforcely - @reyronald has just merged this PR in, so I'm closing out this issue.  Yay!!

**Comment by robbieheywood** [comment]
Hey @reyronald / @RobertChrist  - this change is great! Is it possible to get a new release of the extension that includes this? I think there's only a couple of days until the old PR experience is removed.

**Comment by reyronald** [comment]
@robbieheywood A release has already been pushed to the store and it's in revision process. Should become available soon.

**Comment by chrisaligent** [comment]
Just got to work, opened Chrome, and saw that the extension updated itself to the latest version.
Went to Bitbucket, opened a PR, switched to the new PR experience...and we have syntax highlighting! :tada: 
So far it seems to be working perfectly!

Just here to say thankyou to @reyronald @RobertChrist (and anyone else who may have been involved) for the amazing work, this makes Bitbucket usable again, and just in time for Atlassian to force the new PR experience onto us later today.

If I see any problems I'll open new Github issues (most likely if/when Atlassian break things in upcoming releases), but right now it seems to be very stable.

Thanks again!

**Comment by reyronald** [comment]
Hey @chrisaligent ,

Thanks a lot for stopping by to drop that heartfelt comment!

I always deeply appreciate when users go out of their way to express their gratitude, and makes the effort of maintaining the extension and working on these features much more worthwhile!

I hope we can keep all of you satisfied in the future as well.

Cheers!
